src/jit/codegenxarch.cpp

```diff
@@ -6192,18 +6256,30 @@ void CodeGen::genCallInstruction(GenTreePtr node)
             {
                 assert((arg->gtGetOp1()->OperGet() == GT_PUTARG_STK) && (arg->gtGetOp2()->OperGet() == GT_PUTARG_STK));
             }
+            if ((arg->OperGet() == GT_PUTARG_STK) && (arg->gtGetOp1()->OperGet() == GT_FIELD_LIST))
```

Comment:
Ah, I missed the `else`. Given that the block guarded by the `if (arg->OperGet() == GT_LONG)` is simply an `assert`, perhaps we should wrap the whole statement in `#if DEBUG`; it's your call.